rfc/20241003-temporary-files.md

```diff
@@ -0,0 +1,447 @@
+# Explicit mechanism for working with Temporary Files
+
+Issues: This is indirectly motivated by both [#1086](https://github.com/opentofu/opentofu/issues/1086) and [#1942](https://github.com/opentofu/opentofu/issues/1942), but also represents a distinct feature in its own right.
+
+Although ideally an OpenTofu configuration would do all of its work in local RAM and remote APIs, there are some situations where the most appropriate technique is to write a temporary file to disk and then read that file back into memory in some other location.
+
+For example, when preparing large files to be uploaded into blob storage systems like Amazon S3 it's common to want to prepare the file using some other provider and then ask the `hashicorp/aws` provider to read that file and upload it into an Amazon S3 bucket (or equivalent for other blob storage services).
+
+OpenTofu has no built-in features for managing temporary files, and so informal patterns have emerged. The two most prominent approaches are:
+
+- Use `path.cwd` to generate a path referring to a file in the current working directory, and then pass that absolute path to all of the providers that need to write or read the file in question.
```

Comment:
Worth calling out that there is also `path.root` and per https://opentofu.org/docs/language/expressions/references/#filesystem-and-workspace-info it is recommended over `path.cwd`